src/llm_vm/completion/optimize.py

```diff
@@ -216,14 +215,29 @@ def complete_delay_train(self, stable_context, dynamic_prompt, c_id = None, **kw
             def promiseCompletion():
                 best_completion = self.call_big(prompt, **kwargs)
                 def actual_train(use_completion = None):
+                    def data_synthesis(prompt, response, **kwargs):
+                        final_prompt = '{"prompt": "' +prompt+'"  , "response": "' +response+'" }'+'\nGenerate 100 more JSONS each with a prompt and response field like the given one. The content of the prompt and response fields must be similar to the given JSON. Seperate each JSON with a ,.'
```

Comment:
I'm a bit concerned that the statement "similar to the given JSON" will cause too much repetition.

Reply:
Also, 100 should probably be a variable modifiable by the object

Reply:
When running this as a code this currently generated the task by just changing the country.Couple ways I thought of solving this problem:
1. Separate into "prompt template" (a GPT-3 recognised word) and variables. Then ask the gpt-3 to change specific paramaters for the template.
        Prompt template: What is the [variable] of the [object]? 
                        - variable-object pairs = 1. currency, Myanmar
                                                  2. price, Bitcoin
Now, for generation we might specify differences for either prompt templates or variable/object.Based on following parameters(not comprehensive list):
  i. Style: Keep the meaning of the question same but ask in different styles.
            eg: introduce spelling mistakes, code mixing, cultural tone difference etc
  ii. Semantic diversity: The task itself should change.

2. Add the above nlp metrics in code and set thresholds, keep generating till the thersholds are satisfied.
    Reference: 1. https://arxiv.org/pdf/2005.04118.pdf (behavioral  testing of the nlp)
               2.https://arxiv.org/abs/2111.09963 (behavioral testing of recommnder system)
   essentially we will framing the task as NLP task or recommendation task with behaviour metrics satistfied.
   eg: keep generating similar sentences with cosine_similarity 0.5 and diversity 0.4 till "n" sentences reached or "x" attempts satisfied.

Reply:
> I'm a bit concerned that the statement "similar to the given JSON" will cause too much repetition.

As in not different enough examples, and fine tuning on too similar data examples?

Reply:
> eep the meaning of the question same but ask in different styles.
>                    eg: introduce spelling mistakes, code mixing, cultural tone difference etc.
> 
>        ii. Semantic diversity: The task itself should chang

This seems like a really good idea to try, I will make some tickets